cpp/include/raft/spatial/knn/knn.cuh

```diff
@@ -224,4 +345,84 @@ void brute_force_knn(raft::handle_t const& handle,
                                metric,
                                metric_arg);
 }
+
+/**
+ * @brief Flat C++ API function to perform a brute force knn on
+ * a series of input arrays and combine the results into a single
+ * output array for indexes and distances. Inputs can be either
+ * row- or column-major but the output matrices will always be in
+ * row-major format.
+ *
+ * @example
+ *
+ *
+ *
+ * @param[in] handle the cuml handle to use
+ * @param[in] index vector of device matrices (each size m_i*d) to be used as the knn index
+ * @param[in] search matrix (size n*d) to be used for searching the index
+ * @param[out] indices matrix (size n*k) to store output knn indices
+ * @param[out] distances matrix (size n*k) to store the output knn distance
+ * @param[in] k the number of nearest neighbors to return
+ * @param[in] metric_arg the value of `p` for Minkowski (l-p) distances. This
+ * 					 is ignored if the metric_type is not Minkowski.
```

Comment:
Would you consider combining `metric` and `metric_arg` into a single type, where it's impossible to construct an instance with a `p` value for a non-Minkowski metric?

Reply:
We've adopted this method of specifying a metric w/ an optional metric arg from Scipy and Scikit-learn. This was done intentionally to provide a familiar face for the APIs. There are also a few other places where this design is used, such as pairwise distances and sparse pw dists/knn. I don't mind having a larger discussion on this topic (maybe on a Github issue?) but I'm wondering if we should hold that change for a post 22.10 release.